src/System.Net.Http/tests/FunctionalTests/HttpClientTest.cs

```diff
@@ -545,7 +545,21 @@ private sealed class CustomResponseHandler : HttpMessageHandler
 
             protected override Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
             {
-                return _func(request, cancellationToken);
+                try
+                {
+                    return DisposeWhenCompleted(_func(request, cancellationToken), request);
+                }
+                catch
+                {
+                    request.Content?.Dispose();
+                    throw;
+                }
+            }
+
+            private static async Task<T> DisposeWhenCompleted<T>(Task<T> t, HttpRequestMessage request)
+            {
+                try { return await t; }
+                finally { request.Content?.Dispose(); }
```

Comment:
Does this mean that the job of disposing the content is now always on the caller?  Or is this because we're not disposing HttpRequestMessage until later in the tests?

(And if this requires explicit user Disposing, is it the thing that if you don't dispose you can deadlock your process because IO is waiting on GC is waiting on memory pressure is waiting on IO?)